src/Thoth.Json.Net/Decode.fs

```diff
@@ -615,53 +581,39 @@ module Decode =
             builder { new IGetters with
                 member __.Required =
                     { new IRequiredGetter with
                         member __.Field (fieldName : string) (decoder : Decoder<_>) =
-                            match decodeValueError path (field fieldName decoder) v with
-                            | Ok v -> v
-                            | Error error ->
-                                raise (DecoderException error)
+                            unwrap path (field fieldName decoder) v
                         member __.At (fieldNames : string list) (decoder : Decoder<_>) =
-                            match decodeValueError path (at fieldNames decoder) v with
-                            | Ok v -> v
-                            | Error error ->
-                                raise (DecoderException error)
+                            unwrap path (at fieldNames decoder) v
                         member __.Raw (decoder: Decoder<_>) =
-                            match decodeValueError path decoder v with
-                            | Ok v -> v
-                            | Error error ->
-                                raise (DecoderException error) }
+                            unwrap path decoder v }
                 member __.Optional =
                     { new IOptionalGetter with
                         member __.Field (fieldName : string) (decoder : Decoder<_>) =
-                            match decodeValueError path (field fieldName decoder) v with
-                            | Ok v -> Some v
-                            | Error (_, BadField _ ) -> None
-                            | Error (_, BadType (_, jToken))
-                            | Error (_, BadPrimitive (_, jToken)) when jToken.Type = JTokenType.Null -> None
-                            | Error error ->
-                                raise (DecoderException error)
+                            unwrap path (field fieldName (option decoder)) v
                         member __.At (fieldNames : string list) (decoder : Decoder<_>) =
-                            if Helpers.isObject v then
-                                match decodeValueError path (at fieldNames decoder) v with
-                                | Ok v -> Some v
-                                | Error (_, BadPath _ )
-                                | Error (_, BadTypeAt _) -> None
-                                | Error (_, BadType (_, jToken))
-                                | Error (_, BadPrimitive (_, jToken)) when jToken.Type = JTokenType.Null -> None
-                                | Error error ->
-                                    raise (DecoderException error)
-                            else
-                                raise (DecoderException (path, BadType ("an object", v)))
+                            unwrap path (at fieldNames (option decoder)) v
+                        // REVIEW: I would have preferred to use `unwrap path (option decoder) v`
```

Comment:
I am not sure to understand.

If you don't include `at fieldNames` etc. then the decoders isn't a `At` decoder